### PR TITLE
Added 'skip' options to goals.

### LINKED ISSUE #732: Add `skip` options to goals
### Description

Similar in spirit to the `skip` goals of d-m-p, fabric8 should support options to skip certain goals:

* `resource` (`fabric8.skip.resource`)
* `build` (already supported via `docker.skip.build`, should have a correspondent `fabric8.skip.build`)
* `push` (already supported via `docker.skip.push`, should have a correspondent `fabric8.skip.push`)
* `apply` (`fabric8.skip.apply`)
* `deploy` (`fabric8.skip.deploy`)

### PR DESCRIPTION
**Pull request by dev-gaur**
fixes #732 .

### CODE REVIEW

**Comment by dev-gaur** [comment]
Should I add some tests for this? @rhuss @rohanKanojia ?

**Comment by rohanKanojia** [comment]
It's fine without tests in my opinion

File: plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ApplyMojo.java [hunk 1]
```diff
@@ -193,10 +193,16 @@
     @Parameter(property = "fabric8.s2i.buildNameSuffix", defaultValue = "-s2i")
     protected String s2iBuildNameSuffix;
 
+    @Parameter(property = "docker.skip.apply", defaultValue = "false")
+    protected boolean skipApply;
```
[the remark this answers is not in the document]
Comment (dev-gaur) [reply plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ApplyMojo.java]: oops

**Comment by Ladicek** [comment]
I just found by accident that many options are called `docker.skip.*` instead of `fabric8.skip.*`.

**Comment by rohanKanojia** [comment]
Our Push and Build goals inherit Push/Build goals from DMP. Do you think those options are being propagated here?

**Comment by Ladicek** [comment]
I don't know about that. I see that this pull request includes a bunch of parameters named `docker.skip.*`, even though the classes are in `io.fabric8.maven.plugin`.

**Comment by rohanKanojia** [comment]
ah, yeah. I can see it now :eyeglasses:

**Pull request merged**